privaterelay/tests/mgmt_sync_phone_related_dates_on_profile_tests.py

```diff
@@ -0,0 +1,233 @@
+        date_subscribed_phone,
+        date_subscribed_phone,
+        date_subscribed_phone + timedelta(30),
+    )
+
+    num_profiles_updated = sync_phone_related_dates_on_profile("subscription")
+
+    profile.refresh_from_db()
+    assert profile.date_phone_subscription_reset == date_subscribed_phone
+    assert profile.date_subscribed_phone == date_subscribed_phone
+    assert profile.date_phone_subscription_start == date_subscribed_phone
+    assert profile.date_phone_subscription_end == date_subscribed_phone + timedelta(30)
+    assert num_profiles_updated == 1
+
+
+@patch(f"{MOCK_BASE}.get_phone_subscription_dates")
+def test_monthly_phone_subscriber_renewed_subscription_profile_date_phone_subscription_start_and_end_updated(
+    mocked_dates, patch_datetime_now, phone_user
+):
+    profile = Profile.objects.get(user=phone_user)
+    first_day_of_current_month = datetime.now(timezone.utc).replace(day=1)
+    # get first of the last month
+    first_day_of_last_month = (first_day_of_current_month - timedelta(1)).replace(day=1)
+    # get first of the next month
+    first_day_of_next_month = (first_day_of_current_month + timedelta(31)).replace(
+        day=1
+    )
+    profile.date_subscribed_phone = first_day_of_last_month
+    profile.date_phone_subscription_start = first_day_of_last_month
+    profile.date_phone_subscription_end = first_day_of_current_month
+    # assume that phone limit reset command has not been ran
+    profile.date_phone_subscription_reset = first_day_of_current_month
+    profile.save()
+    mocked_dates.return_value = (
+        first_day_of_last_month,
+        first_day_of_current_month,
+        first_day_of_next_month,
+    )
+
+    num_profiles_updated = sync_phone_related_dates_on_profile("subscription")
+
+    profile.refresh_from_db()
+    # dates stayed same
+    assert profile.date_phone_subscription_reset == first_day_of_current_month
+    assert profile.date_subscribed_phone == first_day_of_last_month
+    # dates updated
+    assert profile.date_phone_subscription_start == first_day_of_current_month
+    assert profile.date_phone_subscription_end == first_day_of_next_month
+    assert num_profiles_updated == 1
+
+
+@patch(f"{MOCK_BASE}.get_phone_subscription_dates")
+def test_yearly_phone_subscriber_profile_date_fields_all_updated(
+    mocked_dates, patch_datetime_now, phone_user
+):
+    profile = Profile.objects.get(user=phone_user)
+    date_subscribed_phone = datetime.now(timezone.utc) - timedelta(3)
+    profile.date_subscribed_phone = date_subscribed_phone
+    profile.save()
+    mocked_dates.return_value = (
+        date_subscribed_phone,
+        date_subscribed_phone,
+        date_subscribed_phone + timedelta(365),
+    )
+
+    num_profiles_updated = sync_phone_related_dates_on_profile("subscription")
+
+    profile.refresh_from_db()
+    assert profile.date_phone_subscription_reset == date_subscribed_phone
+    assert profile.date_subscribed_phone == date_subscribed_phone
```

Comment:
quibble (non-blocking): (just to prove I'm reading everythign line-by-line!) the test is called `...profile_date_fields_all_updated` but this `date_subscribed_phone` value stays the same.

Reply:
Fixed in c09fe8a